src/ZoneServer/Skills/Handlers/Swordsman/Hoplite/ThrowingSpear.cs

```diff
@@ -0,0 +1,198 @@
+			{ 
+				await Task.Delay(hitDelay);
+
+				var pad = new Pad(PadName.ThrouwingSpear_Hoplite33_Pad, caster, skill, new Circle(farPos, 50));
+				pad.Position = farPos;
+				pad.Trigger.UpdateInterval = TimeSpan.FromSeconds(2);
+				pad.Trigger.Destroyed += this.SpearExplosion;
+
+				caster.Map.AddPad(pad);
+			}
+		}
+
+		/// <summary>
+		/// Executes the main attack
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private async void Attack(Skill skill, ICombatEntity caster, ISplashArea splashArea)
+		{
+			var damageDelay = TimeSpan.FromMilliseconds(50);
+			var skillHitDelay = TimeSpan.Zero;
+
+			await Task.Delay(hitDelay);
+
+			var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
+			var hits = new List<SkillHitInfo>();
+
+			foreach (var target in targets.LimitBySDR(caster, skill))
+			{
+				var modifier = SkillModifier.MultiHit(3);
+
+				var skillHitResult = SCR_SkillHit(caster, target, skill, modifier);
+
+				if (skillHitResult.Result == HitResultType.Crit && caster.Components.TryGet<SkillComponent>(out var skills)
+					&& skills.TryGet(SkillId.Hoplite_SharpSpear, out var sharpSpear))
+				{
+					skillHitResult.Damage += skillHitResult.Damage *= (0.1f + sharpSpear.Level * 0.02f);
+				}
+
+				target.TakeDamage(skillHitResult.Damage, caster);
+
+				var skillHit = new SkillHitInfo(caster, target, skill, skillHitResult, damageDelay, skillHitDelay);
+
+				if (caster.Components.TryGet<AbilityComponent>(out var abilities) && abilities.IsActive(AbilityId.Hoplite33))
+				{
+					skillHit.HitEffect = HitEffect.Impact;					
+				} else
+				{
+					skillHit.KnockBackInfo = new KnockBackInfo(caster.Position, target.Position, skill);
+					skillHit.HitInfo.Type = skill.Data.KnockDownHitType;
+					target.Position = skillHit.KnockBackInfo.ToPosition;
+				}
+
+				target.StartBuff(BuffId.Common_Shock, skill.Level, 0, DebuffDuration, caster);
+				
+				hits.Add(skillHit);
+			}
+
+			Send.ZC_SKILL_HIT_INFO(caster, hits);
+		}
+
+
+		/// <summary>
+		/// Called by the spear explosion effect
+		/// </summary>
+		/// <param name="sender"></param>
+		/// <param name="args"></param>
+		private void SpearExplosion(object sender, TriggerArgs args)
+		{
+			if (args.Trigger is not Pad pad) return;
+			if (pad.Creator is not ICombatEntity creator) return;
+			if (pad.Skill is not Skill skill) return;
+
+			this.ExplosionAttack(pad.Skill, creator, (ISplashArea)pad.Area);
+		}
+
+		/// <summary>
+		/// Executes the secondary attack from the spear explosion
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="splashArea"></param>
+		private void ExplosionAttack(Skill skill, ICombatEntity caster, ISplashArea splashArea)
+		{
+			var damageDelay = TimeSpan.FromMilliseconds(50);
+			var skillHitDelay = TimeSpan.Zero;
+
+			var targets = caster.Map.GetAttackableEntitiesIn(caster, splashArea);
```

Comment:
Instead of keep writing custom solutions, I believe it would be preferable to utilize the pad more. You define it with the trigger area, and it tracks the actors inside already. You don't need a second call into `Map`. And if you feel like the pad should be able to return X random attackable actors from inside, why not add that method and see how it feels?

Reply:
For implementations with potentially wide-ranging impacts like this it's always my preference to let you add them.  For example, I don't really know if that method should have the code to override the value if ignoreSDR is set, or if the pad should use its maxActors value to limit how many values it returns.